### PR TITLE
update to msal 4.17.0

### PR DESCRIPTION
**Pull request by jennyf19**
tested w/the dev samples + b2c.

### CODE REVIEW

File: tests/Microsoft.Identity.Web.Test.LabInfrastructure/Microsoft.Identity.Web.Test.LabInfrastructure.csproj [hunk 1]
```diff
@@ -17,7 +17,7 @@
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
-    <PackageReference Include="Microsoft.Identity.Client" Version="4.16.1" />
```
Comment (jmprieur) [inline tests/Microsoft.Identity.Web.Test.LabInfrastructure/Microsoft.Identity.Web.Test.LabInfrastructure.csproj:20]: @jennyf19 @pmaytak 
Given that the `Microsoft.Identity.Web.Test.LabInfrastructure.csproj` project references `Microsoft.Identity.Web.Test.Common.csproj` which itself [includes](https://github.com/AzureAD/microsoft-identity-web/blob/25ff0df874662d92225ccef28ae60c50ce5406c5/tests/Microsoft.Identity.Web.Test.Common/Microsoft.Identity.Web.Test.Common.csproj#L26) the Microsoft.Identity.Web project, MSAL.NET is provided by transitivity, and therefore this package reference is no longer necessary

File: tests/Microsoft.Identity.Web.Test.LabInfrastructure/Microsoft.Identity.Web.Test.LabInfrastructure.csproj [hunk 2]
```diff
@@ -17,7 +17,7 @@
...
+    <PackageReference Include="Microsoft.Identity.Client" Version="4.17.0" />
```
Comment (jmprieur) [inline tests/Microsoft.Identity.Web.Test.LabInfrastructure/Microsoft.Identity.Web.Test.LabInfrastructure.csproj:20]:
```suggestion
```

File: tests/Microsoft.Identity.Web.Test.LabInfrastructure/Microsoft.Identity.Web.Test.LabInfrastructure.csproj (hunk 1 above)
Comment (jennyf19) [reply tests/Microsoft.Identity.Web.Test.LabInfrastructure/Microsoft.Identity.Web.Test.LabInfrastructure.csproj:20]: ok. i've tried taking it out before and that hasn't worked, but i'll try again. thanks for the suggestion.

File: tests/Microsoft.Identity.Web.Test.LabInfrastructure/Microsoft.Identity.Web.Test.LabInfrastructure.csproj (hunk 2 above)
Comment (jennyf19) [reply tests/Microsoft.Identity.Web.Test.LabInfrastructure/Microsoft.Identity.Web.Test.LabInfrastructure.csproj:20]: i'll test this first.

**Pull request merged**